demo/source/index.ts

```diff
@@ -96,16 +72,18 @@ async function deployAllTheThings(rpc: SignerFetchRpc) {
 	} as const
 }
 
-function hexStringToByteArray(hex: string) {
-	const match = /^(?:0x)?([a-fA-F0-9]*)$/.exec(hex)
-	if (match === null) throw new Error(`Expected a hex string encoded byte array with an optional '0x' prefix but received ${hex}`)
-	const normalized = match[1]
-	if (normalized.length % 2) throw new Error(`Hex string encoded byte array must be an even number of charcaters long.`)
-	const bytes = []
-	for (let i = 0; i < normalized.length; i += 2) {
-		bytes.push(Number.parseInt(`${normalized[i]}${normalized[i + 1]}`, 16))
+async function ethGetBlockByNumber(rpc: FetchJsonRpc, blockNumber: bigint): Promise<OracleSdk.Block | null> {
+	const result = await rpc.getBlockByNumber(false, blockNumber)
+	if (result === null) throw new Error(`Unknown block number ${blockNumber}`)
+	if (result.logsBloom === null) throw new Error(`Block ${blockNumber} was missing 'logsBloom' field.`)
+	if (result.number === null) throw new Error(`Block ${blockNumber} was missing 'number' field.`)
+	return {
+		...result,
+		logsBloom: result.logsBloom!,
```

Comment:
Do these need `!` since you're checking their only other union above?

Reply:
Nope, fixed.  Had them in there before I did the asserts above, forgot to remove them.